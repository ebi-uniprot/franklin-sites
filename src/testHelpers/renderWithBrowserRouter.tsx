import { ReactElement } from 'react';
import { BrowserRouter } from 'react-router-dom';
// eslint-disable-next-line import/no-extraneous-dependencies
import { render } from '@testing-library/react';

// Inspired by https://testing-library.com/docs/example-react-router/

const renderWithBrowserRouter = (ui: ReactElement, { route = '/' } = {}) => {
  window.history.pushState({}, 'Test page', route);

  return render(ui, { wrapper: BrowserRouter });
};

export default renderWithBrowserRouter;
