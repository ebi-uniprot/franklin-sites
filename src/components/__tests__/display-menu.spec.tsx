import { screen, fireEvent } from '@testing-library/react';

import DisplayMenu from '../display-menu';

import displayMenuData, {
  displayMenuDummyLeft1,
  displayMenuDummyLeft2,
} from '../__mocks__/displayMenu';

import renderWithBrowserRouter from '../../testHelpers/renderWithBrowserRouter';

describe('Display menu component', () => {
  it('should render', () => {
    const { asFragment } = renderWithBrowserRouter(
      <DisplayMenu data={displayMenuData} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should toggle item content', async () => {
    renderWithBrowserRouter(<DisplayMenu data={displayMenuData} />);
    expect(
      screen
        .getAllByText(displayMenuDummyLeft1)[0]
        .closest('.display-menu__item_content')
    ).toBeInTheDocument();

    expect(screen.queryByText(displayMenuDummyLeft2)).toBeNull();

    fireEvent.click(screen.getByText(displayMenuData[1].name));
    const newContent2 = await screen.findByText(displayMenuDummyLeft2);
    expect(newContent2).toBeInTheDocument();
  });
});
