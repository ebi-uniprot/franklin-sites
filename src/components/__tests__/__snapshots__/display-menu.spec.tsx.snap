// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display menu component should render 1`] = `
<DocumentFragment>
  <ul
    class="display-menu"
  >
    <ul
      class="no-bullet"
    >
      <li>
        <h5
          class="display-menu__item_title"
        >
          <a
            aria-current="page"
            class="display-menu__item_title--active"
            href="/"
          >
            <span
              class="display-menu__item_icon"
            >
              <test-file-stub />
            </span>
            Item 1
          </a>
        </h5>
        <div
          class="display-menu__item_content"
        >
          <ul
            class="no-bullet"
          >
            <li>
              Something
            </li>
            <li>
              Something
            </li>
            <li>
              Something
            </li>
            <li>
              Something
            </li>
            <li>
              Something
            </li>
          </ul>
        </div>
      </li>
      <li>
        <h5
          class="display-menu__item_title"
        >
          <a
            href="///item2"
          >
            <span
              class="display-menu__item_icon"
            />
            Item 2
          </a>
        </h5>
      </li>
      <li>
        <h5
          class="display-menu__item_title"
        >
          <a
            href="///item3"
          >
            <span
              class="display-menu__item_icon"
            />
            Item 3
          </a>
        </h5>
      </li>
    </ul>
  </ul>
</DocumentFragment>
`;
